frontend/components/branchViewLink.jsx

```diff
@@ -3,44 +3,46 @@ import PropTypes from 'prop-types';
 import { connect } from 'react-redux';
 import { SITE } from '../propTypes';
 import { IconView } from './icons';
+import globals from '../globals';
 
-const isDefaultBranch = (branchName, site) => branchName === site.defaultBranch;
-const isDemoBranch = (branchName, site) => branchName === site.demoBranch;
+const getViewLink = (branch, site) => {
+  const {
+    awsBucketName, owner, repository, domains, siteBranchConfigs,
+  } = site;
+  const origin = `https://${awsBucketName}.${globals.PROXY_DOMAIN}`;
+  const branchConfig = siteBranchConfigs.find(s => s.branch === branch);
 
-const getViewText = (branchName, site) => {
-  let viewText = 'Preview site';
-  if (isDefaultBranch(branchName, site)) {
-    viewText = 'View site';
-  } else if (isDemoBranch(branchName, site)) {
-    viewText = 'View demo';
+  if (!branchConfig) {
+    return `${origin}/preview/${owner}/${repository}/${branch}`;
   }
-  return viewText;
-};
 
-export const BranchViewLink = ({
-  branchName, viewLink, site, showIcon,
-}) => {
-  const viewText = getViewText(branchName, site);
-
-  if (showIcon) {
-    return (
-      <a
-        href={viewLink}
-        target="_blank"
-        rel="noopener noreferrer"
-        className="view-site-link"
-      >
-        { viewText }
-        <IconView />
-      </a>
-    );
+  const domain = domains.find(d => d.siteBranchConfigId === branchConfig.id);
+
+  if (!domain || domain.state !== 'provisioned') {
+    return `${origin}${branchConfig.s3Key}`;
   }
-  return (<a href={viewLink} target="_blank" rel="noopener noreferrer">{ viewText }</a>);
+
+  return `https://${domain.names.split(',')[0]}`;
+};
+
+export const BranchViewLink = ({ branchName, site, showIcon }) => {
+  const domain = getViewLink(branchName, site);
+
+  return (
+    <a
+      href={domain}
+      target="_blank"
+      rel="noopener noreferrer"
+      className={showIcon ? 'view-site-link' : ''}
+    >
+      Build Link
```

Comment:
I'd replace with "View Build Link" or something similar (to have the verb match the user action)